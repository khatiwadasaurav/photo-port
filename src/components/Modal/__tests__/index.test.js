import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "../";

afterEach(cleanup);

describe("Modal component", () => {
	const onClose = jest.fn();
	const currentPhoto = {
		name: "test",
		category: "commercial",
		description: "test",
		index: 0,
	};
	it("renders", () => {
		render(<Modal onClose={onClose} currentPhoto={currentPhoto} />);
	});

	it("matches snapshot DOM node structure", () => {
		// Arrange the snapshot - declare variables
		const { container } = render(
			<Modal onClose={onClose} currentPhoto={currentPhoto} />
		);

		// Act - perform the action
		// Assert - check the result
		expect(container.firstChild).toMatchSnapshot();
	});
});

describe("Click Event", () => {
	const onClose = jest.fn();
	const currentPhoto = {
		name: "test",
		category: "commercial",
		description: "test",
		index: 0,
	};
	it("calls onClose handler", () => {
		// Arrange: Render Modal
		const { getByText } = render(
			<Modal onClose={onClose} currentPhoto={currentPhoto} />
		);
		// Act: Simulate click event
		const closeButton = getByText("Close this modal");
		closeButton.click();

		// Assert: Expected matcher
		expect(onClose).toHaveBeenCalled();
		expect(onClose).toHaveBeenCalledTimes(1);
	});
});
