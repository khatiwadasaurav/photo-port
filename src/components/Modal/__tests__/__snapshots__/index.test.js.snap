// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<div
  class="modalBackdrop"
>
  <div
    class="modalContainer"
  >
    <h3
      class="modalTitle"
    >
      test
    </h3>
    <img
      alt="current category"
    />
    <p>
      test
    </p>
    <button
      type="button"
    >
      Close this modal
    </button>
  </div>
</div>
`;
